controllers/agents/svc/agentsvc_controller.go

```diff
@@ -70,6 +72,15 @@ func (r *AgentServiceReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		if err := r.removeServiceClasses(ctx, req); err != nil {
 			return ctrl.Result{}, err
 		}
+
```

Comment:
Not sure if the logic to add listing the service classes is required. We can simply reque if the err is not nil on L72. WDYT?